### DISCUSSION TITLE
Provide a way to customize/prefix Livewire routes

### DISCUSSION THREAD

**Discussion by ragulka** [livewire/livewire#1674] [4. General]
It looks like there's currently no way to customize Livewire routes - they're hard-coded:

https://github.com/livewire/livewire/blob/803024bcdfff299aa296d1256ec2e4b7834b80a8/src/LivewireServiceProvider.php#L129-L141

It would be nice if it was possible to at least prefix/namespace these routes. It would be useful for a multi-tenant application that uses path identification to ensure any eloquent queries inside the LW component are scoped to the current tenant.

Another idea would be to allow adding custom headers to Livewire's AJAX requests, so one could always attach the tenant ID to the headers and identify tenant based on that.

**Comment by kslimani** [comment]
I have the same issue. It seems this feature is required (especially with Jetstream).
If you change the `asset_url` value in the `config/livewire.php` file and rebuild the assets with npm you will get 404 http errors because it does not change livewire routes.

This feature seems really simple to solve, i will attempt to create a pull request to propose 3 new configuration options  : "route_namespace", "route_domain" and "route_prefix" (which defaults to null to keep current behaviour)

**Comment by kslimani** [comment]
What about something like :

```php
        RouteFacade::group([
            'namespace' => config('livewire.route_namespace', null),
            'domain' => config('livewire.route_domain', null),
            'prefix' => config('livewire.route_prefix', null),
        ], function () {
            RouteFacade::post('/livewire/message/{name}', HttpConnectionHandler::class)
                ->name('livewire.message')
                ->middleware(config('livewire.middleware_group', ''));

            RouteFacade::post('/livewire/upload-file', [FileUploadHandler::class, 'handle'])
                ->name('livewire.upload-file')
                ->middleware(config('livewire.middleware_group', ''));

            RouteFacade::get('/livewire/preview-file/{filename}', [FilePreviewHandler::class, 'handle'])
                ->name('livewire.preview-file')
                ->middleware(config('livewire.middleware_group', ''));

            RouteFacade::get('/livewire/livewire.js', [LivewireJavaScriptAssets::class, 'source']);
            RouteFacade::get('/livewire/livewire.js.map', [LivewireJavaScriptAssets::class, 'maps']);
        });
```

It should allow some customization without breaking anything at current behaviour ?

For example, if you change `asset_url` value to `"/some-prefix"` (and rebuild assets using npm) and set `route_prefix` value to `"some-prefix"`, you no longer get 404 errors. (it does not solves everything, but it may be a good start ?)

**Comment by kslimani** [comment]
I have made a pull request #2662 which should fix the issues.
